src/client.rs

```diff
@@ -105,11 +130,17 @@ impl Client {
     }
 
     fn update_rc4_key(&mut self, rc4_key: Vec<u8>) {
-        unimplemented!();
+        self.context.cipher = Rc4::new(&rc4_key.clone());
+        self.context.decipher = Rc4::new(&rc4_key.clone());
     }
```

Comment:
I originally had the skeleton `update_rc4_key` accept a `Vec<u8>` before knowing what the implementation would be, but it looks like we can modify this slightly to avoid some extra memory allocation.

`Rc4::new` takes a reference, so we don't need to clone the value to use it, just provide a reference.
```rs
fn update_rc4_key(&mut self, rc4_key: &[u8]) {
  self.context.cipher = Rc4::new(rc4_key);
  self.context.decipher = Rc4::new(rc4_key);
}
```

Which also means down below we don't need to convert the reset access key to a vector (and allocate space by extension):
```rs
self.update_rc4_key("CD&ML".as_bytes());
```